lib/puppet/functions/nodes_for_role.rb

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+Puppet::Functions.create_function(:nodes_for_role) do
+  dispatch :nodes_for_role do
+    required_param 'String', :role
+    return_type 'Array[String]'
+  end
+
+  def nodes_for_role(role)
+    db_role = role.split('::').map { |x| x.capitalize }.join('::')
+
+    call_function('puppetdb_query',
+                  ['from', 'resources',
+                   ['extract', ['certname'],
+                    ['=', 'title', db_role]]]).map { |x| x['certname'] }
```

Comment:
Could be useful for understanding the result here by naming the `x` parameter to what we're actually iterating over.  I suspect it's a node?  Having that name would give me an idea what to go look for in puppet docs.